core/src/main/scala/cromwell/core/retry/Retry.scala

```diff
@@ -0,0 +1,50 @@
+package cromwell.core.retry
+
+import akka.actor.ActorSystem
+import cromwell.core.CromwellFatalException
+
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.concurrent.{ExecutionContext, Future}
+import akka.pattern.after
+
+object Retry {
+  /**
+    * Retries a Future on a designated backoff strategy until either a designated number of retries or a fatal error
+    * is reached.
+    *
+    * @param f A Future thunk which will be executed once per cycle
```

Comment:
Is this really what you meant? https://en.wikipedia.org/wiki/Thunk

I've never seen that word before!